### PR TITLE
 galerio: init at 1.2.0 

### PR DESCRIPTION
**Pull request by dbrgn**
## Description of changes

This adds a package for [galerio](https://github.com/dbrgn/galerio), a simple generator for self-contained HTML flexbox galleries written in Rust.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dbrgn** [comment]
This is my first package submission. Maybe @fabianhauser or @das-g have time and motivation for a review? 🙂

File: pkgs/by-name/ga/galerio/package.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "galerio";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "dbrgn";
+    repo = "galerio";
+    rev = "v${version}";
+    sha256 = "sha256-JR/YfMUs5IHBRr3uYqHXLNcr23YHyDvgH2y/1ip+2Y8=";
```
Comment (Lurkki14) [inline pkgs/by-name/ga/galerio/package.nix:11]:
```suggestion
    hash = "sha256-JR/YfMUs5IHBRr3uYqHXLNcr23YHyDvgH2y/1ip+2Y8=";
```

File: pkgs/by-name/ga/galerio/package.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  cargoSha256 = "sha256-nYaCN09LP/2MfNRY8oZKtjzFCBFCeRF1IZ2ZBmbHg7I=";
```
Comment (Lurkki14) [inline pkgs/by-name/ga/galerio/package.nix:14]:
```suggestion
  cargoHash = "sha256-nYaCN09LP/2MfNRY8oZKtjzFCBFCeRF1IZ2ZBmbHg7I=";
```

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Comment by Lurkki14** [comment]
`maintainers: Add dbrgn` should be `maintainers: add dbrgn`

**Comment by dbrgn** [comment]
@Lurkki14 all feedback integrated into the branch! Thanks for the review.

**Review by Lurkki14 — APPROVED** [review]

**Review by fabianhauser — APPROVED** [review]

**Review by h7x4 — COMMENTED** [review]
Build and tested on x86_64 nixos, works as expected.

Verified githubId, but I can't verify your gpg key. Could you redo at least the maintainer commit and sign it with your key? https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#reviewing-contributions

Diff looks good otherwise :+1:

**Comment by dbrgn** [comment]
@h7x4 Good point about the PGP signature. Both commits are now signed. (Note that the fingerprint refers to the primary key, not the subkey.)

**Comment by dbrgn** [comment]
@Lurkki14 @fabianhauser @h7x4 the requested changes have been made. It seems that all approvals were cleared by the push. Would you mind re-approving?

Also: Is there a way to get this easily backported to the current NixOS release?

**Comment by h7x4** [comment]
I'll have another look!

The easiest way for backporting is to add the backport label. A bot will automatically cherry-pick the commits and construct a new PR against the chosen branch, if it's able to (if there are conflicts, you'd have to do it manually)

**Review by h7x4 — APPROVED** [review]
Verified the gpg key, looks good!

**Comment by dbrgn** [comment]
Thank you @h7x4!

**Pull request merged**